File: GO/GO_DSA_QS_EASY/PROB_3258_Count_Substrings_That_Satisfy_K_Constraint_I.go
package main

func countKConstraintSubstrings(s string, k int) int {
    cz , co , r , l , a := 0 , 0 , 0 , 0 , 0
    for r < len(s) {
        if(s[r] == '0'){
            cz++
        }else{
            co++
        }
        for co > k && cz > k {
            if(s[l] == '0'){
                cz--
            }else{
                co--
            }
            l++
        }
        a += (r - l) + 1
        r++
    }
    return a
}

File: PYTHON/EASY_PY_LEETCODE/PROB_3258_Count_Substrings_That_Satisfy_K_Constraint_I.py
class Solution:
    def countKConstraintSubstrings(self, s: str, k: int) -> int:
        l,r,co,cz,a = 0,0,0,0,0
        while(r < len(s)):
            if s[r] == '0':
                cz += 1
            else:
                co += 1
            while co > k and cz > k:
                if s[l] == '0':
                    cz -= 1
                else:
                    co -= 1
                l += 1
            a += (r - l) + 1
            r += 1
        return a